### DISCUSSION TITLE
Issue deploying a DDR application to eMMC

### DISCUSSION THREAD

**Discussion by mcergun** [polarfire-soc/polarfire-soc-documentation#38] [How to] [answered]
I've been playing with the hss and hss payload generator but I can't seem to figure out how to create and deploy an SMP application onto the eMMC.

[mss_entry.S](https://github.com/polarfire-soc/platform/blob/main/mpfs_hal/startup_gcc/mss_entry.S#L276) seems to route any `IMAGE_LOADED_BY_BOOTLOADER == 1` applications to `u54_single_hart` function, which just [updates their hartid](https://github.com/polarfire-soc/platform/blob/main/mpfs_hal/startup_gcc/system_startup.c#L224) with `MPFS_HAL_FIRST_HART`

Is this the intended method?

When I compile an application with multiple hart methods (u54_1, u54_2, etc.) and create a payload accordingly, the application only works on a single core, or it doesn't work at all.

How would I compile an application and create the payload if there are multiple harts involved?

**Comment by hughbreslin** [comment, marked as answer]
Hey @mcergun :smile: 

This feature was initially planned for single hart applications, we're having a look into it and we'll get back and let you know what the easiest way to get it working with multiple harts would be. I have a suspicion that updating line 224 of [system_startup.c](https://github.com/polarfire-soc/platform/blob/main/mpfs_hal/startup_gcc/system_startup.c#L224) to:

> hls->my_hart_id = read_csr(mhartid);

would work but we need to test it out to be sure

**Reply by hughbreslin** [reply]
Hi @mcergun Do you have a local ESE you could reach out to and I could share a test version of an updated platform to support what you're trying to do? :)

**Reply by mcergun** [reply]
Until now I couldn't really reply or add comment to this discussion.
Can you explain what local ESE is?

For your first suggestion, I have already tried updating my_hart_id with CSR information but that simple change didn't fix my problem.

**Reply by hughbreslin** [reply]
No problem, I mean do you have a local support engineer? I don't really want to post anything pre-release here but I could share it with you directly through email. 

We have an update for this completed internally and it should be released soon anyway :)

**Reply by mcergun** [reply]
> No problem, I mean do you have a local support engineer? I don't really want to post anything pre-release here but I could share it with you directly through email.
> 
> We have an update for this completed internally and it should be released soon anyway :)

Any unofficial ETA on the release? If that's soon enough I may wait for the release as well.

**Reply by hughbreslin** [reply]
Hopefully next week is the best I can give you?

**Reply by hughbreslin** [reply]
Hi @mcergun a bit late but support is out now in the latest version of the [platform](https://github.com/polarfire-soc/platform/releases/tag/2022.08)

**Reply by rejsys** [reply]
How does this support solve the problem? If a bare metal application does not use the [platform](https://github.com/polarfire-soc/platform/releases/tag/2022.08), what needs to be done to execute an application on all cores using the hss payload?
